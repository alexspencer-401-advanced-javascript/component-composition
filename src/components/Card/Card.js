import React from 'react';
import PropTypes from 'prop-types';

const Card = ({ _id, name, image, gifLink }) => {

  if(!name) null;
  else {
    <header><h3>{name}</h3></header>;
  }

  return (
    <div key={_id}>
      <header><h3>{name}</h3></header>
      <figure>
        <img src={image ? image : gifLink} alt={name} />
      </figure>
      <p>{name} has {_id}</p>
    </div>
  );
};

Card.propTypes = {
  _id: PropTypes.string.isRequired,
  name: PropTypes.string,
  image: PropTypes.string,
  gifLink: PropTypes.string
};

export default Card;
