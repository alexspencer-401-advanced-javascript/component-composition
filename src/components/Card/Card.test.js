import React from 'react';
import { shallow } from 'enzyme';
import Card from './Card';

describe('Card component', () => {
  it('renders Card', () => {
    const wrapper = shallow(<Card _id={'asdgadsg'} />);
    expect(wrapper).toMatchSnapshot();
  });
});
