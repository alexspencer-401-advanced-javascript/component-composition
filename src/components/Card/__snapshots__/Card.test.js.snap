// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card component renders Card 1`] = `
<div
  key="asdgadsg"
>
  <header>
    <h3 />
  </header>
  <figure>
    <img />
  </figure>
  <p>
     has 
    asdgadsg
  </p>
</div>
`;
